Fixing build with GCC-4.5

http://bugs.gentoo.org/show_bug.cgi?id=317507

Patch written by Philipp <sefi@s-e-f-i.de>
--- src/celengine/overlay.h
+++ src/celengine/overlay.h
@@ -12,6 +12,7 @@
 
 #include <string>
 #include <iostream>
+#include <cstdio>
 #include <celtxf/texturefont.h>
 
 
--- src/celmath/mathlib.h
+++ src/celmath/mathlib.h
@@ -44,11 +44,6 @@ 
 template<class T> T radToDeg(T r)
     return r * 180 / static_cast<T>(PI);
 }
 
-template<class T> T abs(T x)
-{
-    return (x < 0) ? -x : x;
-}
-
 template<class T> T square(T x)
 {
     return x * x;
--- src/celengine/star.cpp
+++ src/celengine/star.cpp
@@ -10,6 +10,7 @@
 #include <celmath/mathlib.h>
 #include <cstring>
 #include <cassert>
+#include <cstdio>
 #include "celestia.h"
 #include "astro.h"
 #include "orbit.h"
--- src/celestia/celx_object.cpp
+++ src/celestia/celx_object.cpp
@@ -773,7 +773,7 @@ static int object_mark(lua_State* l)
         markAlpha = 1.0f;
     
     Color markColorAlpha(0.0f, 1.0f, 0.0f, 0.9f);
-    markColorAlpha = Color::Color(markColor, markAlpha);
+    markColorAlpha = Color(markColor, markAlpha);
     
     const char* markLabel = celx.safeGetString(6, WrongType, "Fifth argument to object:mark must be a string");
     if (markLabel == NULL)
